File: DSA/Stack/Stack_Array.c
#include<stdio.h>
#define MAX 5

int push(int x,int *top,int stack[]){
	*top=*top+1;
	stack[*top]=x;
}
int pop(int stack[],int *top){
	int m=stack[*top];
	stack[*top]=NULL;
	*top=*top-1;
	return(m);
}
void display(int *top,int stack[]){
	printf("\nStack :  ");
	for(int i=*top;i!=-1;i--){
		printf("%d ",stack[i]);
	}
	printf("\n");
}
void main(){
	int top=-1,stack[MAX],ch,n;
	while(1){
		printf("\tStack\n------------------------\n1.Push\n2.Pop\n3.Display\t\tEnter choice :  ");
		scanf("%d",&ch);
		switch(ch){
			case 1 : if(top==MAX-1){
				 	 	printf("Stack Overflow.\n");
				 	 	break;
					 }
					 else{
					 	printf("\t\t\tEnter data :  ");
					 	scanf("%d",&n);
					 	push(n,&top,stack);
					 }
					 break;
			case 2 : if(top==-1){
					 	printf("Stack Underflow.\n");
					 }
					 else{
					 	n=pop(stack,&top);
					 	printf("Popped Value :  %d\n",n);
					 }
					 break;
			case 3 : if(top==-1)
						printf("Stack Empty.\n");
					 else
					 	display(&top,stack);
					 break;
			default : exit(0);
		}
	}
}

File: Java/New folder/Stack/Stack_Array.java
package my.dsa;

public class Stack_Array{
    int top=-1;
    int []stack;
    int size;
    public Stack_Array(int size){
        this.size=size;
        stack=new int[size];
    }
    public void push(int data){
        top+=1;
        stack[top]=data;
    }
    public int pop(){
        int popped_value=stack[top];
        top-=1;
        return popped_value;
    }
    public void display(){
        System.out.print("Stack :  ");
        for(int i=top;i>=0;i--){
            System.out.print(stack[i]+" ");
        }
        System.out.println();
    }
    public int top(){
        return stack[top];
    }
    public boolean isEmpty(){
        if(top==-1)return true;
        return false;
    }
    public boolean isFull(){
        if(top==size-1)return true;
        return false;
    }
    public void changeTop(int data){
        stack[top]=data;
    }
}